src/core/SparseMatrix.cpp

```diff
@@ -68,6 +68,15 @@ SparseMatrix::SetZeros(){
     fArmaMat = arma::sp_mat(fNCols, fNRows);
 }
 
+void
+SparseMatrix::SetIdentity(){
+    if(!fNRows && !fNCols && fNCols!=fNRows )
+        return;
+    for (size_t i = 0; i < fNCols; ++i) {
+        fArmaMat(i,i)=1;
```

Comment:
I assumed that because it is a sparse matrix everything that is not accounted for is treated as zero.